src/rpc.ts

```diff
@@ -1,123 +1,78 @@
-import { getAddress } from '@ethersproject/address';
 import { capture } from '@snapshot-labs/snapshot-sentry';
 import express from 'express';
-import { EMPTY_ADDRESS, MAX_STRATEGIES } from './constants';
+import { INVALID_ADDRESS_MESSAGE } from './constants';
 import disabled from './disabled.json';
 import getStrategies from './helpers/strategies';
 import getValidations from './helpers/validations';
-import { getVp, validate } from './methods';
+import { getVp, validate, verifyGetVp, verifyValidate } from './methods';
 import serve from './requestDeduplicator';
 import scores from './scores';
 import {
   blockNumByNetwork,
   checkInvalidStrategies,
   formatStrategies,
+  isAddressValid,
   rpcError,
   rpcSuccess
 } from './utils';
 import { version } from '../package.json';
 
 const router = express.Router();
 
+const METHODS = {
+  get_vp: {
+    verify: verifyGetVp,
+    run: getVp
+  },
+  validate: {
+    verify: verifyValidate,
+    run: validate
+  }
+};
+
+function handlePostError(
+  res: express.Response,
+  params: any,
+  method: string,
+  e: any,
+  id: string | null
+) {
+  capture(e, { params, method });
+  let error = JSON.stringify(e?.message || e || 'Unknown error').slice(0, 1000);
+
+  // Detect provider error
+  if (e?.reason && e?.error?.reason && e?.error?.url) {
+    error = `[provider issue] ${e.error.url}, reason: ${e.reason}, ${e.error.reason}`;
+  }
+
+  console.log(`[rpc] ${method} failed`, JSON.stringify(params), error);
+  return rpcError(res, 500, e, id);
+}
+
 router.post('/', async (req, res) => {
   const { id = null, method, params = {} } = req.body;
 
-  if (!method) return rpcError(res, 400, 'missing method', id);
+  if (params.space && disabled.includes(params.space))
+    return rpcError(res, 429, 'too many requests', id);
 
-  try {
-    if (
-      (method === 'get_vp' && !params.address) ||
-      (method === 'validate' && !params.author) ||
-      params.address === EMPTY_ADDRESS ||
-      params.author === EMPTY_ADDRESS
-    ) {
-      throw new Error('invalid address');
-    }
-    getAddress(params.address || params.author);
-  } catch (e: any) {
-    return rpcError(res, 400, 'invalid address', id);
+  if (!METHODS[method]) {
```

Comment:
Better to extract !METHODS[method] to a variable, hard to read it everywhere, but not important we can ignore